codegen/smithy-aws-go-codegen/src/main/java/software/amazon/smithy/aws/go/codegen/customization/DynamoDBValidateResponseChecksum.java

```diff
@@ -0,0 +1,143 @@
+
+    /**
+     * Gets the sort order of the customization from -128 to 127, with lowest
+     * executed first.
+     *
+     * @return Returns the sort order, defaults to -40.
+     */
+    @Override
+    public byte getOrder() {
+        return 127;
+    }
+
+    @Override
+    public void writeAdditionalFiles(
+            GoSettings settings,
+            Model model,
+            SymbolProvider symbolProvider,
+            GoDelegator goDelegator
+    ) {
+        if (!isDynamoDBService(model, settings.getService(model))) {
+            return;
+        }
+
+        goDelegator.useShapeWriter(settings.getService(model), this::writeMiddlewareHelper);
+    }
+
+    private void writeMiddlewareHelper(GoWriter writer) {
+        writer.openBlock("func $L(stack *middleware.Stack, options Options) {", "}", CHECKSUM_ADDER, () -> {
+            writer.write("$T(stack, $T{Disable: options.$L})",
+                    SymbolUtils.createValueSymbolBuilder(CHECKSUM_INTERNAL_ADDER,
+                            AwsCustomGoDependency.DYNAMODB_CUSTOMIZATION).build(),
+                    SymbolUtils.createValueSymbolBuilder(CHECKSUM_INTERNAL_ADDER + "Options",
+                            AwsCustomGoDependency.DYNAMODB_CUSTOMIZATION).build(),
+                    CHECKSUM_CLIENT_OPTION
+            );
+        });
+        writer.write("");
+
+        writer.openBlock("func $L(stack *middleware.Stack, options Options) {", "}", GZIP_ADDER, () -> {
+            writer.write("$T(stack, $T{Enable: options.$L})",
+                    SymbolUtils.createValueSymbolBuilder(GZIP_INTERNAL_ADDER,
+                            AwsCustomGoDependency.DYNAMODB_CUSTOMIZATION).build(),
+                    SymbolUtils.createValueSymbolBuilder(GZIP_INTERNAL_ADDER + "Options",
+                            AwsCustomGoDependency.DYNAMODB_CUSTOMIZATION).build(),
+                    GZIP_CLIENT_OPTION
+            );
+        });
+        writer.write("");
+    }
+
+    @Override
+    public List<RuntimeClientPlugin> getClientPlugins() {
+        return ListUtils.of(
+                // Add DynamoDB Checksum customization middleware to deserialize.
+                RuntimeClientPlugin.builder()
+                        .servicePredicate(DynamoDBValidateResponseChecksum::isDynamoDBService)
+                        .configFields(ListUtils.of(
+                                AwsConfigField.builder()
+                                        .name(CHECKSUM_CLIENT_OPTION)
+                                        .type(SymbolUtils.createValueSymbolBuilder("bool")
+                                                .putProperty(SymbolUtils.GO_UNIVERSE_TYPE, true)
+                                                .build())
+                                        .documentation("Allows you to disable the client's validation of "
+                                                + "response integrity using CRC32 checksum. Enabled by default.")
+                                        .build()
```

Comment:
So I originally took this approach when I was adding SigV4 authentication. I don't think it makes much sense to use AwsConfigField here. That type was really just meant for defining fields that are present in the global `Config` and mapping them to the client and registering their global default value if not provided. Here your usage is enitrely satisfied by just using the normal ConfigField builder.

Reply:
good point, updated.